nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UpdateRecord.java

```diff
@@ -159,14 +397,19 @@ protected Record process(Record record, final FlowFile flowFile, final ProcessCo
 
             if (evaluateValueAsRecordPath) {
                 final String replacementValue = context.getProperty(recordPathText).evaluateAttributeExpressions(flowFile).getValue();
-                final RecordPath replacementRecordPath = recordPathCache.getCompiled(replacementValue);
-
-                // If we have an Absolute RecordPath, we need to evaluate the RecordPath only once against the Record.
-                // If the RecordPath is a Relative Path, then we have to evaluate it against each FieldValue.
-                if (replacementRecordPath.isAbsolute()) {
-                    record = processAbsolutePath(replacementRecordPath, result.getSelectedFields(), record);
+                if ("".equals(replacementValue)) {
+                    List<FieldValue> selectedFields = result.getSelectedFields().collect(Collectors.toList());
+                    record = remove(record, selectedFields);
                 } else {
-                    record = processRelativePath(replacementRecordPath, result.getSelectedFields(), record);
+                    final RecordPath replacementRecordPath = recordPathCache.getCompiled(replacementValue);
+
+                    // If we have an Absolute RecordPath, we need to evaluate the RecordPath only once against the Record.
+                    // If the RecordPath is a Relative Path, then we have to evaluate it against each FieldValue.
+                    if (replacementRecordPath.isAbsolute()) {
+                        record = processAbsolutePath(replacementRecordPath, result.getSelectedFields(), record);
+                    } else {
+                        record = processRelativePath(replacementRecordPath, result.getSelectedFields(), record);
```

Comment:
I don't think the assignment here to record is necessary because processRelativePath does not modify it.